### ISSUE TITLE
Notification Not showing

### ISSUE THREAD

**Issue by pabloxn** [telerik/kendo-angular#2048]
## Current behavior
 Not displaying notifications when used with other components

## Expected behavior
 I would expect i to appear on top of all other components.

## Environment

Package versions:
"@agm/core": "^1.0.0-beta.5",
    "@angular/animations": "^7.2.1",
    "@angular/cdk": "^7.2.1",
    "@angular/common": "^7.2.1",
    "@angular/compiler": "^7.2.1",
    "@angular/core": "^7.2.1",
    "@angular/fire": "^5.1.1",
    "@angular/flex-layout": "^7.0.0-beta.22",
    "@angular/forms": "^7.2.1",
    "@angular/http": "^7.2.1",
    "@angular/material": "^7.2.1",
    "@angular/material-moment-adapter": "^7.2.1",
    "@angular/platform-browser": "^7.2.1",
    "@angular/platform-browser-dynamic": "^7.2.1",
    "@angular/platform-server": "^7.2.1",
    "@angular/router": "^7.2.1",
    "@covalent/core": "^2.0.0",
    "@covalent/dynamic-forms": "^2.0.0",
    "@datorama/akita": "^1.23.3",
    "@fortawesome/angular-fontawesome": "^0.3.0",
    "@fortawesome/fontawesome-pro": "^5.6.3",
    "@fortawesome/fontawesome-svg-core": "^1.2.12",
    "@fortawesome/free-regular-svg-icons": "^5.6.3",
    "@fortawesome/free-solid-svg-icons": "^5.6.3",
    "@fortawesome/pro-regular-svg-icons": "^5.6.3",
    "@ng-idle/core": "^6.0.0-beta.3",
    "@ng-idle/keepalive": "^6.0.0-beta.3",
    "@progress/kendo-angular-buttons": "^4.3.2",
    "@progress/kendo-angular-charts": "^3.6.0",
    "@progress/kendo-angular-dateinputs": "^3.5.4",
    "@progress/kendo-angular-dialog": "^3.10.1",
    "@progress/kendo-angular-dropdowns": "^3.4.2",
    "@progress/kendo-angular-excel-export": "^2.3.0",
    "@progress/kendo-angular-grid": "^3.11.0",
    "@progress/kendo-angular-inputs": "^3.5.0",
    "@progress/kendo-angular-intl": "^1.6.1",
    "@progress/kendo-angular-l10n": "^1.3.0",
    "@progress/kendo-angular-notification": "^1.2.0",
    "@progress/kendo-angular-pdf-export": "^1.2.0",
    "@progress/kendo-angular-popup": "^2.5.0",
    "@progress/kendo-angular-resize-sensor": "^3.1.2",
    "@progress/kendo-angular-scheduler": "^0.5.0",
    "@progress/kendo-angular-toolbar": "^1.0.0",
    "@progress/kendo-data-query": "^1.5.0",
    "@progress/kendo-date-math": "^1.4.0",
    "@progress/kendo-drawing": "^1.5.8",
    "@progress/kendo-recurrence": "^0.1.2",
    "@progress/kendo-theme-material": "^2.2.0",
    "@types/crypto-js": "^3.1.39",
    "@types/hammerjs": "^2.0.35",
    "@types/lodash": "^4.14.112",
    "core-js": "^2.4.1",
    "crypto-js": "^3.1.9-1",
    "firebase": "^5.7.2",
    "font-awesome": "^4.7.0",
    "fsevents": "^1.1.3",
    "fullcalendar": "^4.0.0-alpha",
    "hammerjs": "^2.0.8",
    "highlight.js": "^9.11.0",
    "jquery": "^3.3.1",
    "lodash": "^4.17.11",
    "moment": "^2.23.0",
    "ngx-avatar": "^3.2.0",
    "primeng": "^6.0.2",
    "quill": "^1.3.6",
    "rxjs": "^6.2.2",
    "rxjs-compat": "^6.3.3",
    "showdown": "1.6.0",
    "tesseract.js": "^1.0.10",
    "tslib": "^1.9.0",
    "zone.js": "^0.8.28"

Browser:

- Chrome (desktop) version 71

**Comment by svetq** [comment]
also referenced in ticket 1384558

**Comment by svetq** [comment]
also referenced in 1384902

**Comment by svetq** [comment]
As a workaround, we could provide the NotificationService within the component, that uses it.

**Comment by kambbado** [comment]
I'm currently experiencing the same issue except is that mine I can get the notification to popup when I launch it from within the component when the component is initialized the first time.

If i destroy the component and instantiate it again the notification is not injected into the **appendTo**  element.

I'm not sure if this is an issue on my part or and issue with Notification.

```
@ViewChild('notificationApplendTo', { read: ViewContainerRef }) public notificationApplendTo: ViewContainerRef;

this.notificationService.show({
      appendTo: this.notificationApplendTo,
      content: 'Success Help Context Saved',
      animation: { type: 'slide', duration: 400 },
      position: { horizontal: 'center', vertical: 'top' },
      type: { style: 'success', icon: true },
      closable: true,
      hideAfter: 100000
    });`

<div class="append-container" #notificationApplendTo></div>

.append-container {
  z-index: 2147483647;
  position: relative;
  width: 50% !important;
  height: 50px;
  margin-left: auto !important;
  margin-right: auto !important;
}

```

**Comment by Irikos** [comment]
Hey @kambbado, any luck in fixing the issue? I have the same problem after trying a workaround similar to yours. It works for a while and then it doesn't anymore.

Or any general fixes found?

**Comment by kambbado** [comment]
> Hey @kambbado, any luck in fixing the issue? I have the same problem after trying a workaround similar to yours. It works for a while and then it doesn't anymore.
> 
> Or any general fixes found?

I ended up using **ngx-bootstrap** alert

```
<alert type="success" *ngIf="showSaveAlert" [dismissOnTimeout]="1000" [dismissible]="true" (onClosed)="closeSaveAlert()" class="alert-container">
  Success
</alert>
```

> alert-container {
  z-index: 2147483647;
  width: 30vw;
  height: 60px;
  margin-left: auto !important;
  margin-right: auto !important;
  position: relative;
  display: flex;
  border: none;
}

```
import { AlertModule } from 'ngx-bootstrap/alert';

// -----------------------------------------------------------------------------
  public showSaveSuccess(): void {
    this.showSaveAlert = true;

    this.cdRef.markForCheck();
  }
  // ---------------------------------------------------------------------------------------------------------
  public closeSaveAlert() {
    this.showSaveAlert = false;
    this.cdRef.markForCheck();
  }
```

It works, I hope this helps you.

**Comment by elena-gancheva** [comment]
Hi @kambbado, here is the reproducible [StackBlitz example](https://stackblitz.com/edit/angular-kendo-notification-service-appendto-bug-rblk2v) of the logged issue with the Notification Component and a navigation with the router. 
Are you experiencing the same problem? If not, could you provide a runnable example, so that we can address the problem?

**Comment by kambbado** [comment]
Hi @elena-gancheva, here is my reproducible [StackBlitz example](https://stackblitz.com/edit/kendo-angular-notification-sample) It works once only.  Let me know if you need more details or want me to try something else.  Thank you.

**Comment by elena-gancheva** [comment]
Hi @kambbado, the animation error is caused by destroying a component view (to which the notification is attached via `appendTo` property) while the notification animation is still in progress. The fix is proposed on a PR and you can check [this issue](https://github.com/telerik/kendo-angular/issues/2305) for its status. As a workaround, the animation can simply be disabled, until the fix is merged.

When destroying a component's view, which is holding the `appendTo` container property, the Notification service should be provided in the component's `providers` array. This way every time the component is destroyed and then initialized again, the Notification service will append notifications on the proper and newly created `appendTo` container. Let me know if this helps. Thank you.

**Comment by elena-gancheva** [comment]
**Update**:
 - The fix of the animation throwing error is available in the latest `dev` version of the package
 - The fix of the notification not showing in `appendTo` container which is destroyed, is available in the latest `dev` version (**no need** for providing the `NotificationService` in the component's `providers` array)

**Comment by Irikos** [comment]
Thanks, @kambbado! I switched to ngx-toastr before seeing your reply and it works quite well, I'm gonna stick to it.

All the best!

**Comment by elena-gancheva** [comment]
Shipped with `v2.0.0`.
